### ISSUE TITLE
Unable to build OTIO on Windows

### ISSUE THREAD

**Issue by JoshBurnell** [PixarAnimationStudios/OpenTimelineIO#622]
Building OTIO on Windows isn't working for me, using the steps in the base [readme](https://github.com/PixarAnimationStudios/OpenTimelineIO) or the [Quickstart](https://github.com/PixarAnimationStudios/OpenTimelineIO/blob/master/docs/tutorials/quickstart.md). @meshula and I worked to get a Windows build working a few weeks ago (Thanks, again, Nick!), but I had gone pretty far off the reservation to make that happen and, attempting to update the build, I'm honestly not sure what I did.

I've gone back now to use the "official" docs and, hence, this Issue.

First, I used a basic `python setup.py install`, but the `setup.py`, as currently in master, triggers an error:
```
CMake Error: The source directory "C:/Users/jburnell/Development/OpenTimelineIO/build/temp.win-amd64-2.7/Release/Release
" does not exist.
Specify --help for usage, or press the help button on the CMake GUI.
Traceback (most recent call last):
  File "setup.py", line 466, in <module>
    **PROJECT_METADATA
  File "Z:\rel\rez\win\third_party\python\2.7.13\python-2.7.13.amd64\lib\distutils\core.py", line 151, in setup
    dist.run_commands()
  File "Z:\rel\rez\win\third_party\python\2.7.13\python-2.7.13.amd64\lib\distutils\dist.py", line 953, in run_commands
    self.run_command(cmd)
  File "Z:\rel\rez\win\third_party\python\2.7.13\python-2.7.13.amd64\lib\distutils\dist.py", line 972, in run_command
    cmd_obj.run()
  File "setup.py", line 140, in run
    install.run(self)
  File "Z:\rel\rez\win\third_party\python\2.7.13\python-2.7.13.amd64\lib\site-packages\setuptools\command\install.py", l
ine 61, in run
    return orig.install.run(self)
  File "Z:\rel\rez\win\third_party\python\2.7.13\python-2.7.13.amd64\lib\distutils\command\install.py", line 563, in run

    self.run_command('build')
  File "Z:\rel\rez\win\third_party\python\2.7.13\python-2.7.13.amd64\lib\distutils\cmd.py", line 326, in run_command
    self.distribution.run_command(command)
  File "Z:\rel\rez\win\third_party\python\2.7.13\python-2.7.13.amd64\lib\distutils\dist.py", line 972, in run_command
    cmd_obj.run()
  File "Z:\rel\rez\win\third_party\python\2.7.13\python-2.7.13.amd64\lib\distutils\command\build.py", line 127, in run
    self.run_command(cmd_name)
  File "Z:\rel\rez\win\third_party\python\2.7.13\python-2.7.13.amd64\lib\distutils\cmd.py", line 326, in run_command
    self.distribution.run_command(command)
  File "Z:\rel\rez\win\third_party\python\2.7.13\python-2.7.13.amd64\lib\distutils\dist.py", line 972, in run_command
    cmd_obj.run()
  File "setup.py", line 185, in run
    self.build()
  File "setup.py", line 223, in build
    possibly_install(rerun_cmake=False)
  File "setup.py", line 68, in possibly_install
    env=env
  File "Z:\rel\rez\win\third_party\python\2.7.13\python-2.7.13.amd64\lib\subprocess.py", line 186, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['cmake', 'build', '.', '--target', 'install', '--config', 'Release']' returned
non-zero exit status 1

```

This is caused by this code in `setup.py`
```
        if platform.system() == "Windows":
            cmake_args, env = compute_cmake_args()
            subprocess.check_call(
                ['cmake', 'build', '.', '--target', 'install', '--config', 'Release'],
                cwd=_ctx.build_temp_dir,
                env=env
            )
```

A quick change to `setup.py` makes the build work:
```
diff --git a/setup.py b/setup.py
index 5dce829..5958c66 100755
--- a/setup.py
+++ b/setup.py
@@ -63,8 +63,8 @@ def possibly_install(rerun_cmake):
         if platform.system() == "Windows":
             cmake_args, env = compute_cmake_args()
             subprocess.check_call(
                 ['cmake', 'build', '.', '--target', 'install', '--config', 'Release'],
-                cwd=_ctx.build_temp_dir,
+                cwd=os.path.join(_ctx.build_temp_dir, ".."),
                 env=env
             )
```

Now, OTIO builds, but the dlls (`opentime.dll` and `opentimelineio.dll`) never get copied out of the Release directory, thus never making it into the OTIO release.

I tried using the `--cxx-install-root=` argument. The value of cxx-install-root gets added into the `cmake_args` via `compute_cmake_args` but, as you can see in the above code snippet, those `cmake_args` never get appended to the `cmake` command.

So, I tried doing this to the `setup.py`:
```
diff --git a/setup.py b/setup.py
index 5dce829..5958c66 100755
--- a/setup.py
+++ b/setup.py
@@ -63,8 +63,8 @@ def possibly_install(rerun_cmake):
         if platform.system() == "Windows":
             cmake_args, env = compute_cmake_args()
             subprocess.check_call(
-                ['cmake', 'build', '.', '--target', 'install', '--config', 'Release'],
-                cwd=_ctx.build_temp_dir,
+                ['cmake', 'build', '.', '--target', 'install', '--config', 'Release'] + cmake_args,
+                cwd=os.path.join(_ctx.build_temp_dir, ".."),
                 env=env
             )
```

Unfortunately, that doesn't appear to copy the dll's out either. The resulting cmake command looks like so:
```
>cd C:\Users\jburnell\Development\OpenTimelineIO\build\temp.win-amd64-2.7\
Release\..

> C:\Users\jburnell\Development\OpenTimelineIO\build\temp.win-amd64-2.7>cmake build . --target install --config Release
-DPYTHON_EXECUTABLE=Z:\rel\rez\win\third_party\python\2.7.13\python-2.7.13.amd64\python.exe -DOTIO_PYTHON_INSTALL:BOOL=O
N -DOTIO_PYTHON_OTIO_DIR=C:\Users\jburnell\Development\OpenTimelineIO\build\lib.win-amd64-2.7\opentimelineio\ -DCMAKE_IN
STALL_PREFIX=C:\OpenTimelineIO.cxx -A x64
-- Selecting Windows SDK version 10.0.18362.0 to target Windows 10.0.17763.
-- Updating git submodules to make sure they are up to date
-- pybind11 v2.3.dev0
-- Configuring done
-- Generating done
-- Build files have been written to: C:/Users/jburnell/Development/OpenTimelineIO/build/temp.win-amd64-2.7/Release
```

The `-DCMAKE_INSTALL_PREFIX` appears to be ignored entirely.

Anyway, I'm stumped, so I'm opening this in the hopes that someone with more Windows/cmake experience might have a hint. I can open a PR for the changes I've described to `setup.py`, but I'm not sure that they're right yet.

**Comment by meshula** [comment]
Thanks for the report, I'll do a clean pull, and investigate.

**Comment by JoshBurnell** [comment]
Thanks! If there's anything I can do, lemme know. I'm pretty deep in the code at the moment.

**Comment by darbyjohnston** [comment]
I think there is a problem with how the CMake args are being composed; change this line in Setup.py:

    ['cmake', 'build', '.', '--target', 'install', '--config', 'Release'],

To:

    ['cmake', '--build', '.', '--target', 'install', '--config', 'Release'],

Then the install works OK on my machine.

**Comment by meshula** [comment]
That's definitely a typo, thanks for the catch.

**Comment by JoshBurnell** [comment]
Ah, good one. Yeah, that makes the build work for me (if I revert my tweak to the `cwd` up above).

Lemme run a few tests and see if everything is working as expected.

**Comment by matandobr** [comment]
I don't know if it's connected but I also get this error when trying building manually with cmake
![image](https://user-images.githubusercontent.com/6709304/70341674-27ae1780-185c-11ea-8de4-cf74c93ba762.png)
(Win10)

**Comment by ssteinbach** [comment]
Addressed by #624
